src/hooks/usePayments.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useState } from "react";
+import { CardInput } from "../components/common/Card/card.type";
+import { cardInfo } from "../constants/Payments";
+import { CardListType } from "../types/payments";
+import { formatNumber, isNumber, monthConverter } from "../utils";
+
+const defaultCardInfo = {
+  title: "클린카드",
+  number: "1111-2222-3333-4444",
+  name: "YUJO",
+  expiry: "10/25",
+  nickname: "법인카드",
+  cvc: "123",
+  password1: "1",
+  password2: "2",
+  backgroundColor: "#94dacd",
+};
+
+const defaultNewCardInfo = {};
+
+export const usePayments = () => {
+  const [step, setStep] = useState<number>(0);
+  const [newCardInfo, setNewCardInfo] = useState<CardInput>(defaultNewCardInfo);
+  const [cardList, setCardList] = useState<CardListType[]>([defaultCardInfo]);
+
+  const validateInput = (input: string, id: string): boolean => {
+    if (
+      id === cardInfo.NUMBER ||
+      id === cardInfo.EXPIRY ||
+      id === cardInfo.CVC ||
+      id === cardInfo.PASSWORD1 ||
+      id === cardInfo.PASSWORD2
+    ) {
+      return isNumber(input);
+    }
+    return true;
+  };
+
+  const formatInput = (
+    input: string,
+    id: string,
+    maxLength: number
+  ): string => {
+    if (input && input.length < maxLength) {
+      switch (id) {
+        case cardInfo.NUMBER:
+          return formatNumber(input, 5);
+        case cardInfo.EXPIRY:
+          if (input.length === 2) {
+            return formatNumber(monthConverter(input), 3, "/");
+          }
+          return formatNumber(input, 3, "/");
+      }
+    }
+    return input;
+  };
+
+  const handleCardInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    let { value, id, maxLength } = e.target;
```

Comment:
만약 `let`을 불가피하게 사용한다면 아래와 같이 작성해서 값이 변경되는지 코드를 읽으며 신경 쓸 변수를 줄이면 좋을거 같아요.

물론 `let`을 사용하지 않는게 가장 좋습니다!
```suggestion
    const { id, maxLength } = e.target;
    let { value } = e.target;
```